src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBufferReader.cs

```diff
@@ -512,43 +453,21 @@ public string ReadUTF8String(int length)
             return value;
         }
 
-        public unsafe void UnsafeReadArray(byte* dst, byte* dstMax)
+        public void ReadRawArrayBytes<T>(Span<T> dst)
+            where T : unmanaged
         {
-            UnsafeReadArray(dst, (int)(dstMax - dst));
+            ReadRawArrayBytes(MemoryMarshal.AsBytes(dst));
         }
 
-        private unsafe void UnsafeReadArray(byte* dst, int length)
+        public void ReadRawArrayBytes(Span<byte> dst)
         {
-            if (_stream != null)
+            if (dst.Length > 0)
             {
-                const int chunk = 256;
-                while (length >= chunk)
-                {
```

Comment:
I removed the chunked read of at most 256bytes at a time for the stream version since I belive that the approach was partly due to reduce risk of DOS attacks where large array sizes are specified, however GetBuffer (actually TryEnsureBytes) now contains a fix with mitigations against that.

If you think it  is better to keep the chunked read I can add it back, but then maybe the size can be increased to something like 4k or even 8k (to almost read a jumbo frame) ?